<?xml version="1.0"?>
<launch>
    <arg name="lidar_frame_id" default="velodyne" />
    <arg name="use_exact_sync" default="false" />
    <arg name="queue_size" default="10" />
    <arg name="map_resolution" default="10" />
    <arg name="input_points_topic" default="/points_filtered" />
    <arg name="input_odom_topic" default="/odom" />
    <arg name="output_map_topic" default="/map" />

    <!-- Node Manager Arguments -->
    <arg name="node_name" value="floam" />
    <arg name="manager_name" value="$(arg node_name)_manager" />
    <arg name="global_manager_name" value="/$(arg manager_name)" />
    <arg name="new_manager" default="false" />

    <!-- Nodelet Manager -->
    <node pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" output="screen" if="$(arg new_manager)" />
  
    <!-- floam mapping node -->
    <node pkg="nodelet" type="nodelet" name="$(arg node_name)_mapping_node"
      args="load floam/lidar/LidarMappingNode $(arg global_manager_name)" output="screen">
      <param name="use_exact_sync" value="$(arg use_exact_sync)" />
      <param name="queue_size" value="$(arg queue_size)" />
      <param name="map_resolution" value="$(arg map_resolution)" />

      <remap from="points_filtered" to="$(arg input_points_topic)" />
      <remap from="odom" to="$(arg input_odom_topic)" />
      <remap from="map" to="$(arg output_map_topic)" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="world2map_tf"  args="0 0 0 0 0 0 /world /map 10" />

  	<node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="gt" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="$(arg lidar_frame_id)" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="base_link" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="base_link" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>

</launch>
